core/chains/evm/logpoller/orm.go

```diff
@@ -240,59 +227,64 @@ type Exp struct {
-	q := o.q.WithOpts(qopts...)
-
-	return q.ExecQ(`WITH r AS
+func (o *orm) DeleteExpiredLogs(ctx context.Context) error {
+	_, err := o.db.ExecContext(ctx, `WITH r AS
 		( SELECT address, event, MAX(retention) AS retention
 			FROM evm.log_poller_filters WHERE evm_chain_id=$1 
 			GROUP BY evm_chain_id,address, event HAVING NOT 0 = ANY(ARRAY_AGG(retention))
 		) DELETE FROM evm.logs l USING r
 			WHERE l.evm_chain_id = $1 AND l.address=r.address AND l.event_sig=r.event
 			AND l.created_at <= STATEMENT_TIMESTAMP() - (r.retention / 10^9 * interval '1 second')`, // retention is in nanoseconds (time.Duration aka BIGINT)
 		ubig.New(o.chainID))
+	return err
 }
 
 // InsertLogs is idempotent to support replays.
-func (o *DbORM) InsertLogs(logs []Log, qopts ...pg.QOpt) error {
+func (o *orm) InsertLogs(ctx context.Context, logs []Log) error {
 	if err := o.validateLogs(logs); err != nil {
 		return err
 	}
-
-	return o.q.WithOpts(qopts...).Transaction(func(tx pg.Queryer) error {
-		return o.insertLogsWithinTx(logs, tx)
+	return o.Transaction(ctx, func(orm *orm) error {
+		return o.insertLogsWithinTx(ctx, logs, orm.db.(*sqlx.Tx))
 	})
 }
 
-func (o *DbORM) InsertLogsWithBlock(logs []Log, block LogPollerBlock, qopts ...pg.QOpt) error {
+func (o *orm) InsertLogsWithBlock(ctx context.Context, logs []Log, block LogPollerBlock) error {
 	// Optimization, don't open TX when there is only a block to be persisted
 	if len(logs) == 0 {
-		return o.InsertBlock(block.BlockHash, block.BlockNumber, block.BlockTimestamp, block.FinalizedBlockNumber, qopts...)
+		return o.InsertBlock(ctx, block.BlockHash, block.BlockNumber, block.BlockTimestamp, block.FinalizedBlockNumber)
 	}
 
 	if err := o.validateLogs(logs); err != nil {
 		return err
 	}
 
 	// Block and logs goes with the same TX to ensure atomicity
-	return o.q.WithOpts(qopts...).Transaction(func(tx pg.Queryer) error {
-		if err := o.InsertBlock(block.BlockHash, block.BlockNumber, block.BlockTimestamp, block.FinalizedBlockNumber, pg.WithQueryer(tx)); err != nil {
+	return o.Transaction(ctx, func(orm *orm) error {
+		if err := o.insertBlockWithinTx(ctx, orm.db.(*sqlx.Tx), block.BlockHash, block.BlockNumber, block.BlockTimestamp, block.FinalizedBlockNumber); err != nil {
 			return err
 		}
-		return o.insertLogsWithinTx(logs, tx)
+		return o.insertLogsWithinTx(ctx, logs, orm.db.(*sqlx.Tx))
 	})
 }
 
-func (o *DbORM) insertLogsWithinTx(logs []Log, tx pg.Queryer) error {
+func (o *orm) insertBlockWithinTx(ctx context.Context, tx *sqlx.Tx, blockHash common.Hash, blockNumber int64, blockTimestamp time.Time, finalizedBlock int64) error {
+	query := `INSERT INTO evm.log_poller_blocks (evm_chain_id, block_hash, block_number, block_timestamp, finalized_block_number, created_at)
+			VALUES ($1, $2, $3, $4, $5, NOW())
+			ON CONFLICT DO NOTHING`
+	_, err := tx.ExecContext(ctx, query, ubig.New(o.chainID), blockHash.Bytes(), blockNumber, blockTimestamp, finalizedBlock)
+	return err
+}
```

Comment:
Couldn't we change this param to be the new `sqlutil.Queryer` so we don't have to use `sqlx` and don't have to cast the `orm.db` in the methods that use these funcs?
```suggestion
func (o *orm) insertBlockWithinTx(ctx context.Context, db sqlutil.Queryer, blockHash common.Hash, blockNumber int64, blockTimestamp time.Time, finalizedBlock int64) error {
	query := `INSERT INTO evm.log_poller_blocks (evm_chain_id, block_hash, block_number, block_timestamp, finalized_block_number, created_at)
			VALUES ($1, $2, $3, $4, $5, NOW())
			ON CONFLICT DO NOTHING`
	_, err := db.ExecContext(ctx, query, ubig.New(o.chainID), blockHash.Bytes(), blockNumber, blockTimestamp, finalizedBlock)
	return err
}
```